packages/cozy-procedures/src/redux/documentsDataSlice.js

```diff
@@ -0,0 +1,116 @@
+import { createSlice } from 'redux-starter-kit'
+import get from 'lodash/get'
+import creditApplicationTemplate from '../templates/creditApplicationTemplate'
+import { AdministrativeProcedure } from 'cozy-doctypes'
+
+const documentsSlice = createSlice({
+  initialState: {
+    data: {
+      /*  payslip: {
+        files: []
+      } */
+    },
+    ui: {
+      /* payslip: {
+        loading: true,
+        error: false
+      } */
+      initiated: false
+    }
+  },
+  slice: 'documents',
+  reducers: {
+    init: (state, action) => {
+      return {
+        data: Object.keys(action.payload).reduce((acc, fieldId) => {
+          acc[fieldId] = {
+            files: []
+          }
+          return acc
+        }, {}),
+        ui: Object.keys(action.payload).reduce((acc, fieldId) => {
+          acc[fieldId] = {
+            loading: false,
+            error: false
+          }
+          return acc
+        }, {})
+      }
+    },
+
+    fetchDocumentLoading: (state, action) => {
+      const { idDoctemplate, loading } = action.payload
+      state.ui[idDoctemplate].loading = loading
+    },
+    fetchDocumentSuccess: (state, action) => {
+      const { idDoctemplate, loading, files } = action.payload
+      state.ui[idDoctemplate].loading = loading
+      state.data[idDoctemplate].files = files
+    },
+    fetchDocumentError: (state, action) => {
+      const { idDoctemplate, error } = action.payload
+      state.ui[idDoctemplate].error = error
+    },
+    setProcedureStatus: (state, action) => {
+      state.ui.initiated = action.payload.initiated
+    }
+  }
+})
+const selectors = {
+  getFiles: state => get(state, [documentsSlice.slice, 'data'], {}),
+  getInitiated: state =>
+    get(state, [documentsSlice.slice, 'ui', ['initiated']], {})
+}
```

Comment:
No selector for `loading`? We never display a spinner while documents are being fetched (like for id document here: https://app.zeplin.io/project/584a6f4e838090371f59518f/screen/5cfac0c72d682519cab68ee7 )